prober/http.go

```diff
@@ -65,9 +67,76 @@ func matchRegularExpressions(reader io.Reader, httpConfig config.HTTPProbe, logg
 	return true
 }
 
+// trace holds timings for a single HTTP roundtrip.
+type roundTripTrace struct {
+	tls           bool
+	start         time.Time
+	dnsDone       time.Time
+	connectDone   time.Time
+	gotConn       time.Time
+	responseStart time.Time
+	end           time.Time
+}
+
+// transport is a custom transport keeping traces for each HTTP roundtrip.
+type transport struct {
+	Transport http.RoundTripper
+	logger    log.Logger
+	traces    []*roundTripTrace
+	current   *roundTripTrace
+}
+
+func newTransport(rt http.RoundTripper, logger log.Logger) *transport {
+	return &transport{
+		Transport: rt,
+		logger:    logger,
+		traces:    []*roundTripTrace{},
+	}
+}
+
+// RoundTrip switches to a new trace, then runs embedded RoundTripper.
+func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	trace := &roundTripTrace{}
+	if req.URL.Scheme == "https" {
+		trace.tls = true
+	}
+	t.current = trace
+	t.traces = append(t.traces, trace)
+	defer func() { trace.end = time.Now() }()
+	return t.Transport.RoundTrip(req)
+}
+
+func (t *transport) DNSStart(_ httptrace.DNSStartInfo) {
+	t.current.start = time.Now()
+}
+func (t *transport) DNSDone(_ httptrace.DNSDoneInfo) {
+	t.current.dnsDone = time.Now()
+}
+func (ts *transport) ConnectStart(_, _ string) {
+	t := ts.current
+	// No DNS resolution, e.g. connecting to IP directly.
```

Comment:
s/e.g.//

e.g. means "for example" and this is the only case where there'd be no DNS